src/main/java/org/terasology/equipment/system/EquipmentEffectsSystem.java

```diff
@@ -0,0 +1,109 @@
+import org.terasology.equipment.component.EquipmentEffectComponent;
+import org.terasology.equipment.event.EquipItemEvent;
+import org.terasology.equipment.event.UnequipItemEvent;
+import org.terasology.logic.health.BeforeDamagedEvent;
+import org.terasology.registry.In;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+
+@RegisterSystem
+public class EquipmentEffectsSystem extends BaseComponentSystem {
+
+    @In
+    private Context context;
+
+    /**
+     * Maps the EquipmentEffectComponents to their corresponding EffectComponents so that
+     * 1. the system knows which EquipmentEffectComponents to look out for
+     * 2. the system knows which EffectComponents to remove from the entity on unequip
+     */
+    private Map<Class, Class<? extends  Component>> effectComponents = new HashMap<>();
+
+    @Override
+    public void initialise() {
+        effectComponents.put(StunEffectComponent.class, StunComponent.class);
+        effectComponents.put(RegenEffectComponent.class, RegenerationComponent.class);
+    }
+
+    @ReceiveEvent
+    public void onEquip(EquipItemEvent event, EntityRef entity, EquipmentComponent eq) {
+        //loops through known EquipmentEffectComponents
+        for (Entry<Class, Class<? extends Component>> entry: effectComponents.entrySet()) {
+            EquipmentEffectComponent eec = (EquipmentEffectComponent) event.getItem().getComponent(entry.getKey());
+            if (eec != null && eec.affectsUser) {
+                applyEffect(eec, entity, entity);
+            }
+        }
+    }
+
+    @ReceiveEvent
+    public void onUnequip(UnequipItemEvent event, EntityRef entity, EquipmentComponent eq) {
+        for (Entry<Class, Class<? extends Component>> entry: effectComponents.entrySet()) {
+            EquipmentEffectComponent eec = (EquipmentEffectComponent) event.getItem().getComponent(entry.getKey());
+            if (eec != null && eec.affectsUser) {
+                entity.removeComponent(entry.getValue());
+            }
+        }
+    }
+
+    @ReceiveEvent
+    public void takingDamage(BeforeDamagedEvent event, EntityRef damageTarget) {
+        EntityRef item = event.getDirectCause();
+        for (Entry<Class, Class<? extends Component>> entry: effectComponents.entrySet()) {
+            EquipmentEffectComponent eec = (EquipmentEffectComponent) item.getComponent(entry.getKey());
+            if (eec != null && eec.affectsEnemies) {
+                applyEffect(eec, event.getInstigator(), damageTarget);
+            }
+        }
+    }
+
+    private void applyEffect(EquipmentEffectComponent eec, EntityRef instigator, EntityRef entity) {
+        AlterationEffect alterationEffect = null;
+        if (eec.getClass() == StunEffectComponent.class) {
+            alterationEffect = new StunAlterationEffect(context);
+        } else if (eec.getClass() == RegenEffectComponent.class) {
+            alterationEffect = new RegenerationAlterationEffect(context);
+        }
```

Comment:
There's a much better way to do this that would also allow extensibility (hint: use the `effectComponents` hash map!)